### DISCUSSION TITLE
Could example run on MacOS?

### DISCUSSION THREAD

**Discussion by dugenkui03** [cilium/ebpf#1104] [Q&A] [answered]
I run example and get error message:  getting original memlock rlimit: unsupported platform darwin/amd64.

Could example run on MacOS, or only on linux?

**Comment by dugenkui03** [comment]
can ebpf be a 'log tool' which used to print arguments and returns of go program method？

developer may forget logger.Info for a method which has bug

**Comment by lmb** [comment, marked as answer]
As stated in the [requirements](https://github.com/cilium/ebpf#requirements), this library only works on Linux.
